preprocess.py

```diff
@@ -0,0 +1,91 @@
+'''
+Crated on June 3, 2020
+Author: Joe Tawea
+
+To Do:
+    try-except to check folder before writing
+'''
+
+import sys
+import pandas as pd
+
+# Import Review-Data
+# Remove duplicate reviewerID and asin since book format (hardcover, paperback)
+# Convert reviewerID and asin to categorical
+# Save the DataFrame to pkl
+
+def preprocess_review(review_file, review_pkl):
+
+    print(f'  Reading {review_file}')
+    try:
+        review_df = pd.read_csv(review_file,
+                                dtype={'asin': object, 'reviewerID':object},
+                                sep=',')
+    except:
+        print(f'Error loading {review_file}\nExit!')
+        sys.exit(1)
+
+    print(f'  Done reading {review_file}')
+
+    review_df.drop_duplicates(subset=['reviewerID','asin'],
+                              keep='first', inplace=True)
+    review_df['userID'] = review_df['reviewerID'].astype('category').cat.codes
+    review_df['productID'] = review_df['asin'].astype('category').cat.codes
+
+    print(f'  Writing {review_pkl}')
+    review_df.to_pickle(review_pkl)
+    print(f'  Done writing {review_pkl}')
+
+# Import Meta-Book-Data in chunksize
+# Drop NaN asin, title, category; bad data
+# Drop duplicate asin
+# Clean-up bad title data
+# Save the DataFrame to pkl
+
+def preprocess_product(product_file,product_pkl):
+
+    chunksize = 100000
+    print(f'  Reading {product_file} in chunksize')
+    try:
+        chunk_df = pd.read_json(product_file, lines=True,
+                                dtype={'asin': object}, chunksize=chunksize)
+    except:
+        print(f'Error loading {product_file}\nExit!')
+        sys.exit(1)
+
+    print(f'  Done reading {product_file} in chunksize')
+
+    columns = ['asin', 'title', 'category']
+    product_df = pd.DataFrame(columns=columns)
+
+    for lines in chunk_df:
+        lines = lines[columns]
+
+        lines.dropna(subset=['asin'], inplace=True)
+        lines.dropna(subset=['title'], inplace=True)
+        lines.dropna(subset=['category'], inplace=True)
```

Comment:
would this work: 
`lines.dropna(subset=["asin", "title", "category"], inplace=True)`

Reply:
A better practice would be selecting for the columns you want explicitly. Imagine this scenario:

Someone comes along and has their own data, but it has an extra column named "row_id". This would break downstream because this code doesn't remove an extra column you didn't expect.

Reply:
In fact, if you explicitly select for these columns, this should be even simpler:
`lines.dropna(inplace=True)`